### PR TITLE
Fix backward compatibility issue of using commitTransaction and rollbackTransaction methods during 5.3 Migration

### PR DESCRIPTION
**Pull request by dushaniw**
## Purpose
Fix NoSuchMethodError: org.wso2.carbon.identity.core.util.IdentityDatabaseUtil.commitTransaction  https://github.com/wso2/product-apim/issues/9049

## Goals
Add commitTransaction and rollbackTransaction JDBC util methods.

## Tested
JDK 1.8 
Ubuntu 19.04
Oracle 12c with API-M 2.0 to 2.6
Postgres 9.6 with API-M 2.0 to 2.6

### CODE REVIEW

**Review by inthirakumaaran — APPROVED** [review]

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/util/JDBCPersistenceUtil.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ */
+package org.wso2.carbon.is.migration.service.v530.util;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+public class JDBCPersistenceUtil {
```
Comment (omindu) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/util/JDBCPersistenceUtil.java:26]: Add a class level comment

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/util/JDBCPersistenceUtil.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+     */
+    public static void rollbackTransaction(Connection dbConnection) {
+
+        try {
+            if (dbConnection != null) {
+                dbConnection.rollback();
+            }
+        } catch (SQLException e1) {
+            log.error("An error occurred while rolling back transactions. ", e1);
+        }
```
Comment (omindu) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/util/JDBCPersistenceUtil.java:43]:
```suggestion
        } catch (SQLException e) {
            log.error("An error occurred while rolling back transactions.", e);
        }
```

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/util/JDBCPersistenceUtil.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+     */
+    public static void commitTransaction(Connection dbConnection) {
+
+        try {
+            if (dbConnection != null) {
+                dbConnection.commit();
+            }
+        } catch (SQLException e1) {
+            log.error("An error occurred while commit transactions. ", e1);
+        }
```
Comment (omindu) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/util/JDBCPersistenceUtil.java:59]:
```suggestion
        } catch (SQLException e) {
            log.error("An error occurred while commit transactions.", e);
        }
```

**Review by omindu — CHANGES_REQUESTED** [review]

**Comment by inthirakumaaran** [comment]
For IS testing
- Migration from IS 5.2 to IS 5.7 
- With 1000 users, 1000 SPs, authentication code flow for 5 mins using [JMeter scripts](https://github.com/wso2/performance-is/tree/single-node-performance/distribution/scripts/jmeter)
- JDK 1.8
- DB: Postgres 9.5.23, MySql 5.7

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/util/JDBCPersistenceUtil.java (hunk 1 above)
Comment (dushaniw) [reply components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v530/util/JDBCPersistenceUtil.java:29]: fixed

**Review by omindu — APPROVED** [review]

**Pull request merged**